taskfiles/utils.yml

```
@@ -141,3 +148,158 @@ tasks:
         \( -iname "*.cpp" -o -iname "*.h" -o -iname "*.hpp" \) \
         -print0 | \
           xargs -0 clang-tidy {{.FLAGS}}
+
+  # ===
+  # CMAKE UTILS
+  # ===
+
+  # Runs CMake's configure and build steps for the given source and build directories.
+  #
+  # @param {string} BUILD_DIR CMake build directory to create.
+  # @param {string} SOURCE_DIR Project source directory containing the CMakeLists.txt file.
+  # @param {string=""} [CMAKE_ARGS] Any additional arguments to pass to CMake's configure step.
+  cmake-config-and-build:
+    label: "{{.TASK}}-{{.SOURCE_DIR}}-{{.BUILD_DIR}}"
+    internal: true
+    vars:
+      CMAKE_ARGS: >-
+        {{default "" .CMAKE_ARGS}}
+    requires:
+      vars: ["BUILD_DIR", "SOURCE_DIR"]
+    cmds:
+      - >-
+        cmake
+        -S "{{.SOURCE_DIR}}"
+        -B "{{.BUILD_DIR}}"
+        {{.CMAKE_ARGS}}
+      - >-
+        cmake
+        --build "{{.BUILD_DIR}}"
+        --parallel
+
+  # Unconditionally runs the CMake install step for the given build directory.
```

```suggestion
  # Runs the CMake install step for the given build directory.
```

Unless you think the unconditionally is important?